## Top Performing, Low Income and Underfunded Schools in NC

All data within this report comes from the The Belk Endowment Educational Attainment Data Repository for North Carolina Public Schools:  https://github.com/jakemdrew/EducationDataNC

North Carolina Public Elementary, Middle, and High School campuses are reviewed to identify top performing, low income schools across the state.  

### Assumptions
* **School Year** - This report covers the 2018-19 school year. 
* **Low Income** - This report measures low income using the percentage of economically disadvantaged students at each public school campus.
* **Top Performing** - Public school campuses are sorted in descending order by School Performance Grade.  We also include each school's eg_score as well.  Each list may be resorted by eg_score to identify "most improved" campuses as well.


This report was prepared by: Jake Drew Ph.D. - jakemdrew@gmail.com

In [1]:
#import required Libraries
import pandas as pd
import numpy as np

#Display all records in reports
pd.set_option('display.max_rows', 1000)

## High Schools

There are 685 public high schools reviewed in this report.  We identified a total of 160 high school campuses with a student body that is greater than 60% economically disadvantaged.  


In [5]:
#Read in public high school data
path="C:/Users/Jake/Documents/GitHub/EducationDataNC/2019/School Datasets/PublicHighSchools2019.csv"
schData=pd.read_csv(path, low_memory=False)

In [6]:
print('Total High Schools: ' + str(len(schData)))
print('')
print('Count of NC High Schools by Percentage of Economically Disadvantaged Students')

ranges = [0,10,20,30,40,50,60,70,80,90,100]
pd.cut(schData['pct_eds'], ranges).value_counts().sort_index()

Total High Schools: 685

Count of NC High Schools by Percentage of Economically Disadvantaged Students


(0, 10]       41
(10, 20]      46
(20, 30]      73
(30, 40]     107
(40, 50]     128
(50, 60]     121
(60, 70]      87
(70, 80]      53
(80, 90]      16
(90, 100]      4
Name: pct_eds, dtype: int64

In [7]:
gt60 = len(schData[schData['pct_eds'] > 60])

print('')
print('Total High Schools > 60% Economically Disadvantaged: ' + str(gt60))


Total High Schools > 60% Economically Disadvantaged: 160


## Top Performing SPG Scores, Low Income High Schools

Public school campuses > 60% economically disadvantaged are sorted in descending order by School Performance Grade. We also include each school's eg_score as well. Each list may be resorted by eg_score to identify "most improved" campuses as well. 

In [8]:
EDS_schData = schData[schData['pct_eds'] > 60]
EDS_Cols = ['county','lea_code','name', 'phone', 'pct_eds', 'spg_score','eg_score']
EDS_schData[EDS_Cols].sort_values('spg_score', ascending=False)           

county lea_code                                      name  \
144    Cumberland   260LEA                 Cross Creek Early College   
578         Surry   860LEA                       Surry Early College   
526       Robeson   780LEA                 PSRC Early College at RCC   
521      Richmond   770LEA               Richmond Early College High   
180        Duplin   310LEA                 Duplin Early College High   
289      Guilford   410LEA                          Academy at Smith   
522      Richmond   770LEA                Richmond 9th Grade Academy   
329          Hyde   480LEA                           Ocracoke School   
150    Cumberland   260LEA                  Reid Ross Classical High   
527       Robeson   780LEA                          Red Springs High   
438    Montgomery   620LEA                      East Montgomery High   
42      Brunswick   100LEA                      North Brunswick High   
360         Jones   520LEA                         Jones Senior High   
297      Guilford       CH            Triad Math and Science Academy   
365           Lee   530LEA                  Southern Lee High School   
113      Cherokee   200LEA                              Andrews High   
291      Guilford   410LEA                    Southern Guilford High   
131      Columbus   240LEA                        West Columbus High   
661         Wayne   960LEA                         Spring Creek High   
281      Guilford   410LEA                   Northeast Guilford High   
326          Hoke   470LEA                          Hoke County High   
679        Yadkin   990LEA                      Boonville Elementary   
374       Lincoln   550LEA                           Lincolnton High   
678        Yadkin   990LEA                      Boonville Elementary   
677        Yadkin   990LEA                      Boonville Elementary   
266      Guilford   410LEA                     Eastern Guilford High   
235        Gaston   360LEA                             Ashbrook High   
129      Columbus   240LEA                       South Columbus High   
152    Cumberland   260LEA                           South View High   
498          Pitt   740LEA                        Ayden-Grifton High   
275      Guilford   410LEA                   High Point Central High   
6        Alamance   010LEA                      Hugh M Cummings High   
37         Bladen   090LEA                          West Bladen High   
517      Randolph   761LEA                             Asheboro High   
128      Columbus   240LEA                        East Columbus High   
557       Sampson   820LEA                                Union High   
146    Cumberland   260LEA                            E E Smith High   
520      Richmond   770LEA                      Richmond Senior High   
484       Pamlico       CH                   Arapahoe Charter School   
130      Columbus   240LEA  CCCA - Fair Bluff /Southeastern Campuses   
133      Columbus   241LEA                           Whiteville High   
425   Mecklenburg       CH                       Sugar Creek Charter   
36         Bladen   090LEA                          East Bladen High   
528       Robeson   780LEA                          Saint Pauls High   
555       Sampson   820LEA                             Lakewood High   
228       Forsyth       CH                 Quality Education Academy   
490        Pender   710LEA                          Heide Trask High   
141    Cumberland   260LEA                         Douglas Byrd High   
85       Caldwell   140LEA                        West Caldwell High   
530       Robeson   780LEA                        Purnell Swett High   
299      Guilford       CH       College Prep and Leadership Academy   
269      Guilford   410LEA                      Penn-Griffin Schools   
265      Guilford   410LEA                       James B Dudley High   
263      Guilford   410LEA                    T Wingate Andrews High   
502          Pitt   740LEA                         

## Top Performing SAT Scores,  Low Income High Schools

Public school campuses > 60% economically disadvantaged are sorted in descending order by average SAT scores.

In [9]:
SAT_schData = schData[schData['pct_eds'] > 60]
SAT_Cols = ['county','lea_code','name', 'phone', 'pct_eds', 'avg_sat_score', 'spg_score','eg_score']
SAT_schData[SAT_Cols].sort_values('avg_sat_score', ascending=False)

county lea_code                                      name  \
144    Cumberland   260LEA                 Cross Creek Early College   
329          Hyde   480LEA                           Ocracoke School   
269      Guilford   410LEA                      Penn-Griffin Schools   
521      Richmond   770LEA               Richmond Early College High   
520      Richmond   770LEA                      Richmond Senior High   
578         Surry   860LEA                       Surry Early College   
130      Columbus   240LEA  CCCA - Fair Bluff /Southeastern Campuses   
557       Sampson   820LEA                                Union High   
374       Lincoln   550LEA                           Lincolnton High   
133      Columbus   241LEA                           Whiteville High   
85       Caldwell   140LEA                        West Caldwell High   
356      Johnston   510LEA                     Smithfield-Selma High   
543         Rowan   800LEA                            Salisbury High   
139        Craven   250LEA                          West Craven High   
661         Wayne   960LEA                         Spring Creek High   
360         Jones   520LEA                         Jones Senior High   
42      Brunswick   100LEA                      North Brunswick High   
365           Lee   530LEA                  Southern Lee High School   
491        Pender   710LEA                               Pender High   
517      Randolph   761LEA                             Asheboro High   
152    Cumberland   260LEA                           South View High   
289      Guilford   410LEA                          Academy at Smith   
275      Guilford   410LEA                   High Point Central High   
129      Columbus   240LEA                       South Columbus High   
369        Lenoir   540LEA                         North Lenoir High   
108       Chatham   190LEA                      Jordan Matthews High   
235        Gaston   360LEA                             Ashbrook High   
182        Duplin   310LEA                          James Kenan High   
266      Guilford   410LEA                     Eastern Guilford High   
502          Pitt   740LEA                           North Pitt High   
555       Sampson   820LEA                             Lakewood High   
36         Bladen   090LEA                          East Bladen High   
6        Alamance   010LEA                      Hugh M Cummings High   
291      Guilford   410LEA                    Southern Guilford High   
236        Gaston   360LEA                        Bessemer City High   
150    Cumberland   260LEA                  Reid Ross Classical High   
131      Columbus   240LEA                        West Columbus High   
560      Scotland   830LEA                      Scotland High School   
37         Bladen   090LEA                          West Bladen High   
281      Guilford   410LEA                   Northeast Guilford High   
525       Robeson   780LEA                     Lumberton Senior High   
509      Randolph   760LEA                     Eastern Randolph High   
438    Montgomery   620LEA                      East Montgomery High   
498          Pitt   740LEA                        Ayden-Grifton High   
523       Robeson   780LEA                             Fairmont High   
562        Stanly   840LEA                            Albemarle High   
297      Guilford       CH            Triad Math and Science Academy   
191        Durham   320LEA                             Hillside High   
541         Rowan   800LEA                          North Rowan High   
326          Hoke   470LEA                          Hoke County High   
527       Robeson   780LEA                          Red Springs High   
490        Pender   710LEA                          Heide Trask High   
245        Gaston   360LEA                          Hunter Huss High   
184        Duplin   310LEA                    Wallace-Rose Hill High   
660         Wayne   960LEA                       So

## Top Performing ACT Scores,  Low Income High Schools

Public school campuses > 60% economically disadvantaged are sorted in descending order by average act_scores.

In [10]:
ACT_schData = schData[schData['pct_eds'] > 60]
ACT_Cols = ['county','lea_code','name', 'phone', 'pct_eds', 'act_score', 'spg_score','eg_score']
ACT_schData[ACT_Cols].sort_values('act_score', ascending=False)  

county lea_code                                      name  \
269      Guilford   410LEA                      Penn-Griffin Schools   
144    Cumberland   260LEA                 Cross Creek Early College   
526       Robeson   780LEA                 PSRC Early College at RCC   
578         Surry   860LEA                       Surry Early College   
329          Hyde   480LEA                           Ocracoke School   
521      Richmond   770LEA               Richmond Early College High   
180        Duplin   310LEA                 Duplin Early College High   
289      Guilford   410LEA                          Academy at Smith   
134      Columbus       CH                            Thomas Academy   
297      Guilford       CH            Triad Math and Science Academy   
484       Pamlico       CH                   Arapahoe Charter School   
113      Cherokee   200LEA                              Andrews High   
490        Pender   710LEA                          Heide Trask High   
365           Lee   530LEA                  Southern Lee High School   
133      Columbus   241LEA                           Whiteville High   
150    Cumberland   260LEA                  Reid Ross Classical High   
184        Duplin   310LEA                    Wallace-Rose Hill High   
235        Gaston   360LEA                             Ashbrook High   
291      Guilford   410LEA                    Southern Guilford High   
152    Cumberland   260LEA                           South View High   
139        Craven   250LEA                          West Craven High   
275      Guilford   410LEA                   High Point Central High   
438    Montgomery   620LEA                      East Montgomery High   
661         Wayne   960LEA                         Spring Creek High   
130      Columbus   240LEA  CCCA - Fair Bluff /Southeastern Campuses   
543         Rowan   800LEA                            Salisbury High   
182        Duplin   310LEA                          James Kenan High   
374       Lincoln   550LEA                           Lincolnton High   
360         Jones   520LEA                         Jones Senior High   
589         Union   900LEA                               Monroe High   
108       Chatham   190LEA                      Jordan Matthews High   
555       Sampson   820LEA                             Lakewood High   
648        Warren       CH               Haliwa-Saponi Tribal School   
498          Pitt   740LEA                        Ayden-Grifton High   
517      Randolph   761LEA                             Asheboro High   
453          Nash       CH                   Rocky Mount Preparatory   
369        Lenoir   540LEA                         North Lenoir High   
85       Caldwell   140LEA                        West Caldwell High   
528       Robeson   780LEA                          Saint Pauls High   
525       Robeson   780LEA                     Lumberton Senior High   
266      Guilford   410LEA                     Eastern Guilford High   
356      Johnston   510LEA                     Smithfield-Selma High   
206     Edgecombe       CH                  North East Carolina Prep   
491        Pender   710LEA                               Pender High   
27          Avery       CH                 Marjorie Williams Academy   
541         Rowan   800LEA                          North Rowan High   
42      Brunswick   100LEA                      North Brunswick High   
425   Mecklenburg       CH                       Sugar Creek Charter   
502          Pitt   740LEA                           North Pitt High   
326          Hoke   470LEA                          Hoke County High   
36         Bladen   090LEA                          East Bladen High   
527       Robeson   780LEA                          Red Springs High   
236        Gaston   360LEA                        Bessemer City High   
530       Robeson   780LEA                        Purnell Swett High   
37         Bladen   090LEA                         

## Top Performing Graduation Rates,  Low Income High Schools

Public school campuses > 60% economically disadvantaged are sorted in descending order by average four year graduation rates.

In [11]:
GR4_schData = schData[schData['pct_eds'] > 60]
GR4_Cols = ['county','lea_code','name', 'phone', 'pct_eds', 'pct_STD_ALL_CGR', 'spg_score','eg_score']
GR4_schData[GR4_Cols].sort_values('pct_STD_ALL_CGR', ascending=False)

county lea_code                                      name  \
144    Cumberland   260LEA                 Cross Creek Early College   
227       Forsyth   340LEA            Winston-Salem Preparatory Acad   
526       Robeson   780LEA                 PSRC Early College at RCC   
150    Cumberland   260LEA                  Reid Ross Classical High   
113      Cherokee   200LEA                              Andrews High   
438    Montgomery   620LEA                      East Montgomery High   
521      Richmond   770LEA               Richmond Early College High   
180        Duplin   310LEA                 Duplin Early College High   
289      Guilford   410LEA                          Academy at Smith   
269      Guilford   410LEA                      Penn-Griffin Schools   
578         Surry   860LEA                       Surry Early College   
261        Greene   400LEA                       Greene Central High   
484       Pamlico       CH                   Arapahoe Charter School   
360         Jones   520LEA                         Jones Senior High   
36         Bladen   090LEA                          East Bladen High   
329          Hyde   480LEA                           Ocracoke School   
425   Mecklenburg       CH                       Sugar Creek Charter   
374       Lincoln   550LEA                           Lincolnton High   
491        Pender   710LEA                               Pender High   
490        Pender   710LEA                          Heide Trask High   
228       Forsyth       CH                 Quality Education Academy   
529       Robeson   780LEA                        South Robeson High   
528       Robeson   780LEA                          Saint Pauls High   
37         Bladen   090LEA                          West Bladen High   
291      Guilford   410LEA                    Southern Guilford High   
297      Guilford       CH            Triad Math and Science Academy   
365           Lee   530LEA                  Southern Lee High School   
129      Columbus   240LEA                       South Columbus High   
128      Columbus   240LEA                        East Columbus High   
356      Johnston   510LEA                     Smithfield-Selma High   
133      Columbus   241LEA                           Whiteville High   
601         Vance   910LEA                           AdVance Academy   
498          Pitt   740LEA                        Ayden-Grifton High   
517      Randolph   761LEA                             Asheboro High   
523       Robeson   780LEA                             Fairmont High   
266      Guilford   410LEA                     Eastern Guilford High   
281      Guilford   410LEA                   Northeast Guilford High   
509      Randolph   760LEA                     Eastern Randolph High   
408   Mecklenburg   600LEA                          Rocky River High   
235        Gaston   360LEA                             Ashbrook High   
263      Guilford   410LEA                    T Wingate Andrews High   
139        Craven   250LEA                          West Craven High   
562        Stanly   840LEA                            Albemarle High   
108       Chatham   190LEA                      Jordan Matthews High   
85       Caldwell   140LEA                        West Caldwell High   
369        Lenoir   540LEA                         North Lenoir High   
275      Guilford   410LEA                   High Point Central High   
530       Robeson   780LEA                        Purnell Swett High   
151    Cumberland   260LEA                        Seventy-First High   
541         Rowan   800LEA                          North Rowan High   
42      Brunswick   100LEA                      North Brunswick High   
146    Cumberland   260LEA                            E E Smith High   
453          Nash       CH                   Rocky Mount Preparatory   
229       Forsyth       CH                   Carter G Woodson School   
555       Sampson   820LEA                         

## Top Performing Math 1 Scores,  Low Income High Schools

Public school campuses > 60% economically disadvantaged are sorted in descending order by Math I scores.

In [12]:
M1_schData = schData[schData['pct_eds'] > 60]
M1_Cols = ['county','lea_code','name', 'phone', 'pct_eds', 'm1_score', 'spg_score','eg_score']
M1_schData[M1_Cols].sort_values('m1_score', ascending=False)

county lea_code                                      name  \
269      Guilford   410LEA                      Penn-Griffin Schools   
144    Cumberland   260LEA                 Cross Creek Early College   
521      Richmond   770LEA               Richmond Early College High   
578         Surry   860LEA                       Surry Early College   
526       Robeson   780LEA                 PSRC Early College at RCC   
180        Duplin   310LEA                 Duplin Early College High   
289      Guilford   410LEA                          Academy at Smith   
522      Richmond   770LEA                Richmond 9th Grade Academy   
438    Montgomery   620LEA                      East Montgomery High   
297      Guilford       CH            Triad Math and Science Academy   
326          Hoke   470LEA                          Hoke County High   
329          Hyde   480LEA                           Ocracoke School   
113      Cherokee   200LEA                              Andrews High   
425   Mecklenburg       CH                       Sugar Creek Charter   
229       Forsyth       CH                   Carter G Woodson School   
555       Sampson   820LEA                             Lakewood High   
150    Cumberland   260LEA                  Reid Ross Classical High   
131      Columbus   240LEA                        West Columbus High   
228       Forsyth       CH                 Quality Education Academy   
557       Sampson   820LEA                                Union High   
36         Bladen   090LEA                          East Bladen High   
42      Brunswick   100LEA                      North Brunswick High   
291      Guilford   410LEA                    Southern Guilford High   
130      Columbus   240LEA  CCCA - Fair Bluff /Southeastern Campuses   
360         Jones   520LEA                         Jones Senior High   
484       Pamlico       CH                   Arapahoe Charter School   
374       Lincoln   550LEA                           Lincolnton High   
275      Guilford   410LEA                   High Point Central High   
490        Pender   710LEA                          Heide Trask High   
491        Pender   710LEA                               Pender High   
141    Cumberland   260LEA                         Douglas Byrd High   
206     Edgecombe       CH                  North East Carolina Prep   
527       Robeson   780LEA                          Red Springs High   
128      Columbus   240LEA                        East Columbus High   
498          Pitt   740LEA                        Ayden-Grifton High   
560      Scotland   830LEA                      Scotland High School   
129      Columbus   240LEA                       South Columbus High   
235        Gaston   360LEA                             Ashbrook High   
517      Randolph   761LEA                             Asheboro High   
661         Wayne   960LEA                         Spring Creek High   
139        Craven   250LEA                          West Craven High   
85       Caldwell   140LEA                        West Caldwell High   
133      Columbus   241LEA                           Whiteville High   
184        Duplin   310LEA                    Wallace-Rose Hill High   
365           Lee   530LEA                  Southern Lee High School   
502          Pitt   740LEA                           North Pitt High   
356      Johnston   510LEA                     Smithfield-Selma High   
281      Guilford   410LEA                   Northeast Guilford High   
37         Bladen   090LEA                          West Bladen High   
182        Duplin   310LEA                          James Kenan High   
27          Avery       CH                 Marjorie Williams Academy   
245        Gaston   360LEA                          Hunter Huss High   
408   Mecklenburg   600LEA                          Rocky River High   
146    Cumberland   260LEA                            E E Smith High   
266      Guilford   410LEA                     East

## Top Performing English 2 Scores,  Low Income High Schools

Public school campuses > 60% economically disadvantaged are sorted in descending order by English II scores.

In [13]:
E2_schData = schData[schData['pct_eds'] > 60]
E2_Cols = ['county','lea_code','name', 'phone', 'pct_eds', 'e2_score', 'spg_score','eg_score']
E2_schData[E2_Cols].sort_values('e2_score', ascending=False) 

county lea_code                                      name  \
521      Richmond   770LEA               Richmond Early College High   
144    Cumberland   260LEA                 Cross Creek Early College   
578         Surry   860LEA                       Surry Early College   
269      Guilford   410LEA                      Penn-Griffin Schools   
329          Hyde   480LEA                           Ocracoke School   
526       Robeson   780LEA                 PSRC Early College at RCC   
180        Duplin   310LEA                 Duplin Early College High   
289      Guilford   410LEA                          Academy at Smith   
425   Mecklenburg       CH                       Sugar Creek Charter   
130      Columbus   240LEA  CCCA - Fair Bluff /Southeastern Campuses   
360         Jones   520LEA                         Jones Senior High   
484       Pamlico       CH                   Arapahoe Charter School   
133      Columbus   241LEA                           Whiteville High   
297      Guilford       CH            Triad Math and Science Academy   
42      Brunswick   100LEA                      North Brunswick High   
131      Columbus   240LEA                        West Columbus High   
150    Cumberland   260LEA                  Reid Ross Classical High   
152    Cumberland   260LEA                           South View High   
374       Lincoln   550LEA                           Lincolnton High   
129      Columbus   240LEA                       South Columbus High   
266      Guilford   410LEA                     Eastern Guilford High   
517      Randolph   761LEA                             Asheboro High   
113      Cherokee   200LEA                              Andrews High   
139        Craven   250LEA                          West Craven High   
229       Forsyth       CH                   Carter G Woodson School   
557       Sampson   820LEA                                Union High   
146    Cumberland   260LEA                            E E Smith High   
523       Robeson   780LEA                             Fairmont High   
37         Bladen   090LEA                          West Bladen High   
365           Lee   530LEA                  Southern Lee High School   
27          Avery       CH                 Marjorie Williams Academy   
527       Robeson   780LEA                          Red Springs High   
128      Columbus   240LEA                        East Columbus High   
291      Guilford   410LEA                    Southern Guilford High   
356      Johnston   510LEA                     Smithfield-Selma High   
326          Hoke   470LEA                          Hoke County High   
108       Chatham   190LEA                      Jordan Matthews High   
39         Bladen       CH           Paul R Brown Leadership Academy   
151    Cumberland   260LEA                        Seventy-First High   
509      Randolph   760LEA                     Eastern Randolph High   
281      Guilford   410LEA                   Northeast Guilford High   
408   Mecklenburg   600LEA                          Rocky River High   
438    Montgomery   620LEA                      East Montgomery High   
661         Wayne   960LEA                         Spring Creek High   
155    Cumberland   260LEA                             Westover High   
235        Gaston   360LEA                             Ashbrook High   
228       Forsyth       CH                 Quality Education Academy   
453          Nash       CH                   Rocky Mount Preparatory   
555       Sampson   820LEA                             Lakewood High   
543         Rowan   800LEA                            Salisbury High   
141    Cumberland   260LEA                         Douglas Byrd High   
265      Guilford   410LEA                       James B Dudley High   
191        Durham   320LEA                             Hillside High   
245        Gaston   360LEA                          Hunter Huss High   
36         Bladen   090LEA                         

## Top Performing Biology Scores,  Low Income High Schools

Public school campuses > 60% economically disadvantaged are sorted in descending order by Biology scores.

In [14]:
BIO_schData = schData[schData['pct_eds'] > 60]
BIO_Cols = ['county','lea_code','name', 'phone', 'pct_eds', 'bi_score', 'spg_score','eg_score']
BIO_schData[BIO_Cols].sort_values('bi_score', ascending=False)

county lea_code                                      name  \
144    Cumberland   260LEA                 Cross Creek Early College   
526       Robeson   780LEA                 PSRC Early College at RCC   
578         Surry   860LEA                       Surry Early College   
269      Guilford   410LEA                      Penn-Griffin Schools   
329          Hyde   480LEA                           Ocracoke School   
150    Cumberland   260LEA                  Reid Ross Classical High   
521      Richmond   770LEA               Richmond Early College High   
180        Duplin   310LEA                 Duplin Early College High   
6        Alamance   010LEA                      Hugh M Cummings High   
527       Robeson   780LEA                          Red Springs High   
281      Guilford   410LEA                   Northeast Guilford High   
289      Guilford   410LEA                          Academy at Smith   
42      Brunswick   100LEA                      North Brunswick High   
425   Mecklenburg       CH                       Sugar Creek Charter   
360         Jones   520LEA                         Jones Senior High   
528       Robeson   780LEA                          Saint Pauls High   
133      Columbus   241LEA                           Whiteville High   
297      Guilford       CH            Triad Math and Science Academy   
228       Forsyth       CH                 Quality Education Academy   
85       Caldwell   140LEA                        West Caldwell High   
498          Pitt   740LEA                        Ayden-Grifton High   
131      Columbus   240LEA                        West Columbus High   
365           Lee   530LEA                  Southern Lee High School   
129      Columbus   240LEA                       South Columbus High   
374       Lincoln   550LEA                           Lincolnton High   
520      Richmond   770LEA                      Richmond Senior High   
356      Johnston   510LEA                     Smithfield-Selma High   
152    Cumberland   260LEA                           South View High   
236        Gaston   360LEA                        Bessemer City High   
235        Gaston   360LEA                             Ashbrook High   
113      Cherokee   200LEA                              Andrews High   
291      Guilford   410LEA                    Southern Guilford High   
438    Montgomery   620LEA                      East Montgomery High   
245        Gaston   360LEA                          Hunter Huss High   
661         Wayne   960LEA                         Spring Creek High   
146    Cumberland   260LEA                            E E Smith High   
206     Edgecombe       CH                  North East Carolina Prep   
326          Hoke   470LEA                          Hoke County High   
490        Pender   710LEA                          Heide Trask High   
555       Sampson   820LEA                             Lakewood High   
37         Bladen   090LEA                          West Bladen High   
108       Chatham   190LEA                      Jordan Matthews High   
130      Columbus   240LEA  CCCA - Fair Bluff /Southeastern Campuses   
453          Nash       CH                   Rocky Mount Preparatory   
36         Bladen   090LEA                          East Bladen High   
560      Scotland   830LEA                      Scotland High School   
128      Columbus   240LEA                        East Columbus High   
299      Guilford       CH       College Prep and Leadership Academy   
517      Randolph   761LEA                             Asheboro High   
184        Duplin   310LEA                    Wallace-Rose Hill High   
509      Randolph   760LEA                     Eastern Randolph High   
523       Robeson   780LEA                             Fairmont High   
369        Lenoir   540LEA                         North Lenoir High   
529       Robeson   780LEA                        South Robeson High   
139        Craven   250LEA                         

## Funding data for 2018-2019 is not available yet.  
* I will update the report when NCDPI releases it.

## Top Performing, Lowest Per Pupil Funded Districts - High School Level

This list calculates average per pupil funding at the district level by totaling:
* lea_local_perpupil_num
* lea_state_perpupil_num
* lea_federal_perpupil_num

SPG and Evaas Score averages are taken at the High, Middle, and Elementary School levels.  

Unfortunately, this funding is not available at the individual school level. Individual school level funding values are blank in the Funding table provided by NCDPI. 

Public school campuses are sorted in descending order by School Performance Grade. We also include each school's eg_score as well. Each list may be resorted by eg_score to identify "most improved" campuses as well. 

In [15]:
#Sum up total per pupil school funds
#schData['lea_total_perpupil'] = schData['lea_federal_perpupil'] + \
#                                    schData['lea_local_perpupil'] + \
#                                    schData['lea_state_perpupil']
#Get district average perpupil funds, spg_score, and evaas score 
#distData = schData.groupby('lea_code')['lea_total_perpupil','spg_score','eg_score'].mean()
#distData.reset_index(level=0, inplace=True)

#Divide total funding per pupil into 10 deciles
#print('Count of NC Districts by Totals Funds Per Student at High School Level ')
#pd.cut(distData['lea_total_perpupil'], 10).value_counts().sort_index()

## Top Performing Districts with Lowest Average Per Pupil Funding - High School Level 

In [16]:
#Find top performing districts in the lowest funding decile.
#Funds_distData = distData[distData['lea_total_perpupil'] < 9197.24]
#Funds_distData.sort_values('spg_score', ascending=False) 

In [17]:
#Read in public middle school data
path="C:/Users/Jake/Documents/GitHub/EducationDataNC/2019/School Datasets/PublicMiddleSchools2019.csv"
schData=pd.read_csv(path, low_memory=False)

## Middle Schools

There are 839 public middle schools reviewed in this report.  We identified a total of 314 middle school campuses with a student body that is greater than 60% economically disadvantaged.  


In [18]:
print('Total Middle Schools: ' + str(len(schData)))
print('')
print('Count of NC Middle Schools by Percentage of Economically Disadvantaged Students')

ranges = [0,10,20,30,40,50,60,70,80,90,100]
pd.cut(schData['pct_eds'], ranges).value_counts().sort_index()

Total Middle Schools: 839

Count of NC Middle Schools by Percentage of Economically Disadvantaged Students


(0, 10]       55
(10, 20]      57
(20, 30]      60
(30, 40]      83
(40, 50]      97
(50, 60]     166
(60, 70]     153
(70, 80]     108
(80, 90]      45
(90, 100]      8
Name: pct_eds, dtype: int64

In [19]:
gt60 = len(schData[schData['pct_eds'] > 60])

print('')
print('Total Middle Schools > 60% Economically Disadvantaged: ' + str(gt60))


Total Middle Schools > 60% Economically Disadvantaged: 314


## Top Performing SPG Scores, Low Income Middle Schools

Public school campuses  > 60% economically disadvantaged are sorted in descending order by School Performance Grade. We also include each school's Growth Score as well. Each list may be resorted by eg_score to identify "most improved" campuses. 

In [20]:
EDS_schData = schData[schData['pct_eds'] > 60]
EDS_Cols = ['county','lea_code','name', 'phone', 'pct_eds', 'spg_score','eg_score']
EDS_schData[EDS_Cols].sort_values('spg_score', ascending=False)

county lea_code                                      name  \
374          Hyde   480LEA                           Ocracoke School   
91       Caldwell   140LEA                      Collettsville School   
829        Yadkin   990LEA                       Courtney Elementary   
830        Yadkin   990LEA                       Courtney Elementary   
824        Wilson       CH                    Sallie B Howard School   
342      Guilford       CH            Triad Math and Science Academy   
105      Carteret   160LEA    Down East Middle and Smyrna Elementary   
37         Bladen   090LEA              Clarkton School of Discovery   
96       Caldwell   140LEA                    Kings Creek Elementary   
159        Craven   250LEA                      H J MacDonald Middle   
104      Carteret   160LEA                            Newport Middle   
692         Surry   860LEA                            Central Middle   
129      Cherokee   200LEA                            Andrews Middle   
826        Yadkin   990LEA                      Boonville Elementary   
828        Yadkin   990LEA                      Boonville Elementary   
488   Mecklenburg   600LEA                  Oaklawn Language Academy   
827        Yadkin   990LEA                      Boonville Elementary   
157      Columbus       CH                   Columbus Charter School   
487   Mecklenburg   600LEA                  Oaklawn Language Academy   
693         Surry   860LEA                       J Sam Gentry Middle   
148      Columbus   240LEA                    Cerro Gordo Elementary   
233        Durham       CH                Maureen Joy Charter School   
154      Columbus   241LEA                            Central Middle   
629      Randolph   761LEA                     South Asheboro Middle   
47      Brunswick   100LEA                                  Waccamaw   
278      Franklin   350LEA                               Bunn Middle   
207        Duplin   310LEA                     Chinquapin Elementary   
435       Lincoln   550LEA                         Lincolnton Middle   
177    Cumberland   260LEA                Reid Ross Classical Middle   
633      Richmond   770LEA                             Hamlet Middle   
418         Jones   520LEA                              Jones Middle   
373          Hoke   470LEA                        Sandy Grove Middle   
101      Carteret   160LEA                           Beaufort Middle   
673       Sampson   820LEA                            Hobbton Middle   
590       Pamlico       CH                   Arapahoe Charter School   
832        Yadkin   990LEA                          Starmount Middle   
702  Transylvania   880LEA                             Rosman Middle   
210        Duplin   310LEA                    Kenansville Elementary   
364     Henderson   450LEA                          Flat Rock Middle   
211        Duplin   310LEA                    Kenansville Elementary   
288        Gaston   360LEA                           Holbrook Middle   
349       Halifax   421LEA                    Chaloner Middle School   
630      Randolph   761LEA                     North Asheboro Middle   
281      Franklin   350LEA                       Terrell Lane Middle   
273       Forsyth       CH                 Quality Education Academy   
506   Mecklenburg       CH                       Sugar Creek Charter   
653    Rockingham   790LEA                         J E Holmes Middle   
533         Moore   630LEA                              Elise Middle   
15      Alleghany   030LEA                    Glade Creek Elementary   
73          Burke   120LEA                   Walter R Johnson Middle   
635      Richmond   770LEA                         Rockingham Middle   
55       Buncombe   110LEA                               Enka Middle   
28       Beaufort   070LEA                        Chocowinity Middle   
632      Richmond   770LEA                            Ellerbe Middle   
46      Brunswick   100LEA                    South

## Top Performing Math SPG Scores,  Low Income Middle Schools

Public school campuses > 60% economically disadvantaged are sorted in descending order by Math SPG Scores.

In [21]:
M_schData = schData[schData['pct_eds'] > 60]
M_cols = ['county','lea_code','name', 'phone', 'pct_eds', 'ma_spg_score','eg_score']
M_schData[M_cols].sort_values('ma_spg_score', ascending=False)

county lea_code                                      name  \
374          Hyde   480LEA                           Ocracoke School   
91       Caldwell   140LEA                      Collettsville School   
829        Yadkin   990LEA                       Courtney Elementary   
830        Yadkin   990LEA                       Courtney Elementary   
488   Mecklenburg   600LEA                  Oaklawn Language Academy   
487   Mecklenburg   600LEA                  Oaklawn Language Academy   
824        Wilson       CH                    Sallie B Howard School   
827        Yadkin   990LEA                      Boonville Elementary   
828        Yadkin   990LEA                      Boonville Elementary   
826        Yadkin   990LEA                      Boonville Elementary   
105      Carteret   160LEA    Down East Middle and Smyrna Elementary   
692         Surry   860LEA                            Central Middle   
211        Duplin   310LEA                    Kenansville Elementary   
210        Duplin   310LEA                    Kenansville Elementary   
342      Guilford       CH            Triad Math and Science Academy   
233        Durham       CH                Maureen Joy Charter School   
148      Columbus   240LEA                    Cerro Gordo Elementary   
157      Columbus       CH                   Columbus Charter School   
96       Caldwell   140LEA                    Kings Creek Elementary   
273       Forsyth       CH                 Quality Education Academy   
104      Carteret   160LEA                            Newport Middle   
59       Buncombe   110LEA                        Eblen Intermediate   
154      Columbus   241LEA                            Central Middle   
651       Robeson       CH                               CIS Academy   
153      Columbus   240LEA                         Williams Township   
288        Gaston   360LEA                           Holbrook Middle   
159        Craven   250LEA                      H J MacDonald Middle   
73          Burke   120LEA                   Walter R Johnson Middle   
633      Richmond   770LEA                             Hamlet Middle   
364     Henderson   450LEA                          Flat Rock Middle   
240        Durham       CH                   Global Scholars Academy   
129      Cherokee   200LEA                            Andrews Middle   
435       Lincoln   550LEA                         Lincolnton Middle   
590       Pamlico       CH                   Arapahoe Charter School   
533         Moore   630LEA                              Elise Middle   
203         Davie   300LEA                        South Davie Middle   
629      Randolph   761LEA                     South Asheboro Middle   
373          Hoke   470LEA                        Sandy Grove Middle   
101      Carteret   160LEA                           Beaufort Middle   
693         Surry   860LEA                       J Sam Gentry Middle   
47      Brunswick   100LEA                                  Waccamaw   
207        Duplin   310LEA                     Chinquapin Elementary   
441         Macon   560LEA                Mountain View Intermediate   
349       Halifax   421LEA                    Chaloner Middle School   
834        Yadkin   990LEA                    Yadkinville Elementary   
673       Sampson   820LEA                            Hobbton Middle   
37         Bladen   090LEA              Clarkton School of Discovery   
15      Alleghany   030LEA                    Glade Creek Elementary   
833        Yadkin   990LEA                    Yadkinville Elementary   
506   Mecklenburg       CH                       Sugar Creek Charter   
28       Beaufort   070LEA                        Chocowinity Middle   
149      Columbus   240LEA                      Evergreen Elementary   
597        Pender   710LEA                      Penderlea Elementary   
363     Henderson   450LEA                       Apple Valley Middle   
278      Franklin   350LEA                         

## Top Performing Reading SPG Scores,  Low Income Middle Schools

Public school campuses > 60% economically disadvantaged are sorted in descending order by Reading SPG Scores.

In [22]:
MR_schData = schData[schData['pct_eds'] > 60]
MR_cols = ['county','lea_code','name', 'phone', 'pct_eds', 'rd_spg_score','eg_score']
MR_schData[MR_cols].sort_values('rd_spg_score', ascending=False)

county lea_code                                      name  \
374          Hyde   480LEA                           Ocracoke School   
91       Caldwell   140LEA                      Collettsville School   
37         Bladen   090LEA              Clarkton School of Discovery   
342      Guilford       CH            Triad Math and Science Academy   
96       Caldwell   140LEA                    Kings Creek Elementary   
830        Yadkin   990LEA                       Courtney Elementary   
829        Yadkin   990LEA                       Courtney Elementary   
565   New Hanover       CH    Girls Leadership Academy of Wilmington   
590       Pamlico       CH                   Arapahoe Charter School   
159        Craven   250LEA                      H J MacDonald Middle   
824        Wilson       CH                    Sallie B Howard School   
104      Carteret   160LEA                            Newport Middle   
105      Carteret   160LEA    Down East Middle and Smyrna Elementary   
154      Columbus   241LEA                            Central Middle   
693         Surry   860LEA                       J Sam Gentry Middle   
177    Cumberland   260LEA                Reid Ross Classical Middle   
157      Columbus       CH                   Columbus Charter School   
702  Transylvania   880LEA                             Rosman Middle   
148      Columbus   240LEA                    Cerro Gordo Elementary   
629      Randolph   761LEA                     South Asheboro Middle   
488   Mecklenburg   600LEA                  Oaklawn Language Academy   
673       Sampson   820LEA                            Hobbton Middle   
281      Franklin   350LEA                       Terrell Lane Middle   
487   Mecklenburg   600LEA                  Oaklawn Language Academy   
28       Beaufort   070LEA                        Chocowinity Middle   
435       Lincoln   550LEA                         Lincolnton Middle   
692         Surry   860LEA                            Central Middle   
364     Henderson   450LEA                          Flat Rock Middle   
278      Franklin   350LEA                               Bunn Middle   
344      Guilford       CH       College Prep and Leadership Academy   
506   Mecklenburg       CH                       Sugar Creek Charter   
597        Pender   710LEA                      Penderlea Elementary   
207        Duplin   310LEA                     Chinquapin Elementary   
418         Jones   520LEA                              Jones Middle   
826        Yadkin   990LEA                      Boonville Elementary   
150      Columbus   240LEA                          Hallsboro Middle   
160        Craven   250LEA                           Havelock Middle   
129      Cherokee   200LEA                            Andrews Middle   
422           Lee   530LEA                      SanLee Middle School   
827        Yadkin   990LEA                      Boonville Elementary   
300        Graham   380LEA                       Robbinsville Middle   
101      Carteret   160LEA                           Beaufort Middle   
828        Yadkin   990LEA                      Boonville Elementary   
55       Buncombe   110LEA                               Enka Middle   
423           Lee   530LEA                           West Lee Middle   
351       Halifax       CH                 KIPP Halifax College Prep   
373          Hoke   470LEA                        Sandy Grove Middle   
448      McDowell   590LEA               East McDowell Middle School   
553   New Hanover   650LEA                      Holly Shelter Middle   
46      Brunswick   100LEA                    South Brunswick Middle   
15      Alleghany   030LEA                    Glade Creek Elementary   
635      Richmond   770LEA                         Rockingham Middle   
572        Onslow   670LEA               Jacksonville Commons Middle   
832        Yadkin   990LEA                          Starmount Middle   
317      Guilford   410LEA                      Pen

## Funding data for 2018-2019 is not available yet.  
* I will update the report when NCDPI releases it.

## Top Performing, Lowest Per Pupil Funded Districts - Middle School Level

This list calculates average per pupil funding at the district level by totaling:
* lea_local_perpupil_num
* lea_state_perpupil_num
* lea_federal_perpupil_num

SPG and Evaas Score averages are taken at the High, Middle, and Elementary School levels.  

Unfortunately, this funding is not available at the individual school level. Individual school level funding values are blank in the Funding table provided by NCDPI. 

Public school campuses are sorted in descending order by School Performance Grade. We also include each school's eg_score as well. Each list may be resorted by eg_score to identify "most improved" campuses as well. 

In [23]:
#Sum up total per pupil school funds
#schData['lea_total_perpupil'] = schData['lea_federal_perpupil'] + \
#                                    schData['lea_local_perpupil'] + \
#                                    schData['lea_state_perpupil']
#Get district average perpupil funds, spg_score, and evaas score 
#distData = schData.groupby('lea_code')['lea_total_perpupil','spg_score','eg_score'].mean()
#distData.reset_index(level=0, inplace=True)

#Divide total funding per pupil into 10 deciles
#print('Count of NC Districts by Totals Funds Per Student at Middle School Level ')
#pd.cut(distData['lea_total_perpupil'], 10).value_counts().sort_index()

## Top Performing Districts with Lowest Average Per Pupil Funding - Middle School Level 

In [24]:
#Find top performing districts in the lowest funding decile.
#Funds_distData = distData[distData['lea_total_perpupil'] < 9197.25]
#Funds_distData.sort_values('spg_score', ascending=False)

## Elementary Schools

There are 1636 public elementary schools reviewed in this report.  We identified a total of 651 elementary school campuses with a student body that is greater than 60% economically disadvantaged.  

In [25]:
#Read in public elementary school data
path="C:/Users/Jake/Documents/GitHub/EducationDataNC/2019/School Datasets/PublicElementarySchools2019.csv"
schData=pd.read_csv(path, low_memory=False)

In [26]:
print('Total Elementary Schools: ' + str(len(schData)))
print( '')
print( 'Count of NC Elementary Schools by Percentage of Economically Disadvantaged Students')

ranges = [0,10,20,30,40,50,60,70,80,90,100]
pd.cut(schData['pct_eds'], ranges).value_counts().sort_index()

Total Elementary Schools: 1636

Count of NC Elementary Schools by Percentage of Economically Disadvantaged Students


(0, 10]       76
(10, 20]      92
(20, 30]     105
(30, 40]     147
(40, 50]     262
(50, 60]     297
(60, 70]     315
(70, 80]     224
(80, 90]      87
(90, 100]     25
Name: pct_eds, dtype: int64

In [27]:
gt60 = len(schData[schData['pct_eds'] > 60])

print('')
print('Total Elementary Schools > 60% Economically Disadvantaged: ' + str(gt60))


Total Elementary Schools > 60% Economically Disadvantaged: 651


## Top Performing  SPG Scores, Low Income Elementary Schools

Public school campuses > 60% economically disadvantaged  are sorted in descending order by School Performance Grade. We also include each school's eg_score as well. Each list may be resorted by eg_score to identify "most improved" campuses as well.

In [28]:
EDS_schData = schData[schData['pct_eds'] > 60]
EDS_Cols = ['county','lea_code','name', 'phone', 'pct_eds', 'spg_score','eg_score']
EDS_schData[EDS_Cols].sort_values('spg_score', ascending=False)  

county lea_code                                      name  \
1290       Sampson   820LEA                        Clement Elementary   
701      Henderson   450LEA                    Clear Creek Elementary   
726           Hyde   480LEA                           Ocracoke School   
1360  Transylvania   880LEA                             T C Henderson   
166       Caldwell   140LEA                      Collettsville School   
1218       Robeson   780LEA                      East Robeson Primary   
1359  Transylvania   880LEA                         Rosman Elementary   
1341         Surry   860LEA                         Dobson Elementary   
124          Burke   120LEA                          Icard Elementary   
1346         Surry   860LEA                       Rockford Elementary   
290         Craven   250LEA                Graham A Barden Elementary   
808         Lenoir   540LEA                      Moss Hill Elementary   
190       Carteret   160LEA                        Newport Elementary   
276       Columbus   240LEA                       Old Dock Elementary   
1336        Stokes   850LEA                   Sandy Ridge  Elementary   
340     Cumberland   260LEA                   J W Seabrook Elementary   
790          Jones   520LEA                  Pollocksville Elementary   
1304      Scotland   830LEA                    Laurel Hill Elementary   
1291       Sampson   820LEA                       Hargrove Elementary   
182       Carteret   160LEA                       Atlantic Elementary   
183       Carteret   160LEA                       Atlantic Elementary   
1381         Union   900LEA                         Sardis Elementary   
1296       Sampson   820LEA                          Union Elementary   
277       Columbus   240LEA                       Old Dock Elementary   
54        Beaufort   070LEA                       Chocowinity Primary   
47           Avery   060LEA                  Freedom Trail Elementary   
1615        Yadkin   990LEA                       Courtney Elementary   
186       Carteret   160LEA                 Harkers Island Elementary   
703      Henderson   450LEA                     Edneyville Elementary   
360           Dare   280LEA           Cape Hatteras Elementary School   
798            Lee   530LEA                      Greenwood Elementary   
129          Burke   120LEA                   Ray Childers Elementary   
44            Ashe   050LEA                       Westwood Elementary   
1339         Surry   860LEA                    Cedar Ridge Elementary   
1610        Wilson       CH                    Sallie B Howard School   
187       Carteret   160LEA                 Harkers Island Elementary   
1292       Sampson   820LEA                        Hobbton Elementary   
1616        Yadkin   990LEA                       Courtney Elementary   
1187      Randolph   760LEA                       Archdale Elementary   
811         Lenoir   540LEA                      Pink Hill Elementary   
1137        Pender   710LEA                      Cape Fear Elementary   
254      Cleveland   230LEA                           East Elementary   
1238       Robeson   780LEA                     Tanglewood Elementary   
1334        Stokes   850LEA                       Pinnacle Elementary   
1207      Randolph   761LEA                  Guy B Teachey Elementary   
1352         Surry   862LEA                        Jones Intermediate   
191       Carteret   160LEA    Down East Middle and Smyrna Elementary   
1332        Stokes   850LEA                 Nancy Reynolds Elementary   
281       Columbus   241LEA                       Edgewood Elementary   
1101        Onslow   670LEA                     Summersill Elementary   
793          Jones   520LEA                        Trenton Elementary   
655       Guilford       CH            Triad Math and Science Academy   
43            Ashe   050LEA                  Mountain View Elementary   
261      Cleveland   230LEA                          North Elementary 

## Top Performing Math SPG Scores,  Low Income Elementary Schools

Public school campuses > 60% economically disadvantaged are sorted in descending order by Math SPG scores.

In [29]:
E_schData = schData[schData['pct_eds'] > 60]
E_cols = ['county','lea_code','name', 'phone', 'pct_eds', 'ma_spg_score','eg_score']
E_schData[E_cols].sort_values('ma_spg_score', ascending=False)

county lea_code                                      name  \
1336        Stokes   850LEA                   Sandy Ridge  Elementary   
1381         Union   900LEA                         Sardis Elementary   
1296       Sampson   820LEA                          Union Elementary   
701      Henderson   450LEA                    Clear Creek Elementary   
1218       Robeson   780LEA                      East Robeson Primary   
1290       Sampson   820LEA                        Clement Elementary   
1360  Transylvania   880LEA                             T C Henderson   
290         Craven   250LEA                Graham A Barden Elementary   
254      Cleveland   230LEA                           East Elementary   
726           Hyde   480LEA                           Ocracoke School   
1238       Robeson   780LEA                     Tanglewood Elementary   
1359  Transylvania   880LEA                         Rosman Elementary   
182       Carteret   160LEA                       Atlantic Elementary   
718           Hoke   470LEA                 J W McLauchlin Elementary   
1291       Sampson   820LEA                       Hargrove Elementary   
54        Beaufort   070LEA                       Chocowinity Primary   
183       Carteret   160LEA                       Atlantic Elementary   
1332        Stokes   850LEA                 Nancy Reynolds Elementary   
1346         Surry   860LEA                       Rockford Elementary   
1207      Randolph   761LEA                  Guy B Teachey Elementary   
277       Columbus   240LEA                       Old Dock Elementary   
1341         Surry   860LEA                         Dobson Elementary   
276       Columbus   240LEA                       Old Dock Elementary   
516       Franklin   350LEA                    Edward Best Elementary   
190       Carteret   160LEA                        Newport Elementary   
1338        Stokes   850LEA                    Walnut Cove Elementary   
340     Cumberland   260LEA                   J W Seabrook Elementary   
309     Cumberland   260LEA                  College Lakes Elementary   
1339         Surry   860LEA                    Cedar Ridge Elementary   
1292       Sampson   820LEA                        Hobbton Elementary   
790          Jones   520LEA                  Pollocksville Elementary   
798            Lee   530LEA                      Greenwood Elementary   
1399         Vance   910LEA                         Clarke Elementary   
44            Ashe   050LEA                       Westwood Elementary   
1340         Surry   860LEA                       Copeland Elementary   
1616        Yadkin   990LEA                       Courtney Elementary   
1615        Yadkin   990LEA                       Courtney Elementary   
129          Burke   120LEA                   Ray Childers Elementary   
289         Craven   250LEA            Vanceboro-Farm Life Elementary   
360           Dare   280LEA           Cape Hatteras Elementary School   
166       Caldwell   140LEA                      Collettsville School   
703      Henderson   450LEA                     Edneyville Elementary   
808         Lenoir   540LEA                      Moss Hill Elementary   
1385         Union   900LEA                          Union Elementary   
931    Mecklenburg   600LEA                  Oaklawn Language Academy   
1289       Sampson   820LEA                       Roseboro Elementary   
1187      Randolph   760LEA                       Archdale Elementary   
1304      Scotland   830LEA                    Laurel Hill Elementary   
1137        Pender   710LEA                      Cape Fear Elementary   
930    Mecklenburg   600LEA                  Oaklawn Language Academy   
124          Burke   120LEA                          Icard Elementary   
1213      Richmond   770LEA                Mineral Springs Elementary   
269       Columbus   240LEA                     Acme Delco Elementary   
43            Ashe   050LEA                  Mountain View Elementary 

## Top Performing Reading SPG Scores,  Low Income Elementary Schools

Public school campuses > 60% economically disadvantaged are sorted in descending order by Reading SPG scores.

In [30]:
ER_schData = schData[schData['pct_eds'] > 60]
ER_cols = ['county','lea_code','name', 'phone', 'pct_eds', 'rd_spg_score','eg_score']
ER_schData[ER_cols].sort_values('rd_spg_score', ascending=False) 

county lea_code                                      name  \
1290       Sampson   820LEA                        Clement Elementary   
187       Carteret   160LEA                 Harkers Island Elementary   
186       Carteret   160LEA                 Harkers Island Elementary   
726           Hyde   480LEA                           Ocracoke School   
1359  Transylvania   880LEA                         Rosman Elementary   
1360  Transylvania   880LEA                             T C Henderson   
166       Caldwell   140LEA                      Collettsville School   
701      Henderson   450LEA                    Clear Creek Elementary   
290         Craven   250LEA                Graham A Barden Elementary   
808         Lenoir   540LEA                      Moss Hill Elementary   
790          Jones   520LEA                  Pollocksville Elementary   
791          Jones   520LEA                        Comfort Elementary   
360           Dare   280LEA           Cape Hatteras Elementary School   
1346         Surry   860LEA                       Rockford Elementary   
190       Carteret   160LEA                        Newport Elementary   
277       Columbus   240LEA                       Old Dock Elementary   
54        Beaufort   070LEA                       Chocowinity Primary   
276       Columbus   240LEA                       Old Dock Elementary   
1341         Surry   860LEA                         Dobson Elementary   
793          Jones   520LEA                        Trenton Elementary   
124          Burke   120LEA                          Icard Elementary   
185       Carteret   160LEA                       Beaufort Elementary   
1292       Sampson   820LEA                        Hobbton Elementary   
182       Carteret   160LEA                       Atlantic Elementary   
47           Avery   060LEA                  Freedom Trail Elementary   
183       Carteret   160LEA                       Atlantic Elementary   
850       McDowell   590LEA                Old Fort Elementary School   
261      Cleveland   230LEA                          North Elementary   
184       Carteret   160LEA                    Bogue Sound Elementary   
340     Cumberland   260LEA                   J W Seabrook Elementary   
792          Jones   520LEA                      Maysville Elementary   
655       Guilford       CH            Triad Math and Science Academy   
1616        Yadkin   990LEA                       Courtney Elementary   
1615        Yadkin   990LEA                       Courtney Elementary   
44            Ashe   050LEA                       Westwood Elementary   
173       Caldwell   140LEA                    Kings Creek Elementary   
172       Caldwell   140LEA                         Hudson Elementary   
1304      Scotland   830LEA                    Laurel Hill Elementary   
1218       Robeson   780LEA                      East Robeson Primary   
1101        Onslow   670LEA                     Summersill Elementary   
1128       Pamlico       CH                   Arapahoe Charter School   
798            Lee   530LEA                      Greenwood Elementary   
1137        Pender   710LEA                      Cape Fear Elementary   
1187      Randolph   760LEA                       Archdale Elementary   
43            Ashe   050LEA                  Mountain View Elementary   
703      Henderson   450LEA                     Edneyville Elementary   
1334        Stokes   850LEA                       Pinnacle Elementary   
191       Carteret   160LEA    Down East Middle and Smyrna Elementary   
1610        Wilson       CH                    Sallie B Howard School   
1291       Sampson   820LEA                       Hargrove Elementary   
1238       Robeson   780LEA                     Tanglewood Elementary   
129          Burke   120LEA                   Ray Childers Elementary   
1343         Surry   860LEA                       Franklin Elementary   
168       Caldwell   140LEA                  Dudley Shoals Elementary 

## Funding data for 2018-2019 is not available yet.  
* I will update the report when NCDPI releases it.

## Top Performing, Lowest Per Pupil Funded Districts - Elementary School Level

This list calculates average per pupil funding at the district level by totaling:
* lea_local_perpupil_num
* lea_state_perpupil_num
* lea_federal_perpupil_num

SPG and Evaas Score averages are taken at the High, Middle, and Elementary School levels.  

Unfortunately, this funding is not available at the individual school level. Individual school level funding values are blank in the Funding table provided by NCDPI. 

Public school campuses are sorted in descending order by School Performance Grade. We also include each school's eg_score as well. Each list may be resorted by eg_score to identify "most improved" campuses as well. 

In [31]:
#Sum up total per pupil school funds
#schData['lea_total_perpupil'] = schData['lea_federal_perpupil'] + \
#                                    schData['lea_local_perpupil'] + \
#                                    schData['lea_state_perpupil']
#Get district average perpupil funds, spg_score, and evaas score 
#distData = schData.groupby('lea_code')['lea_total_perpupil','spg_score','eg_score'].mean()
#distData.reset_index(level=0, inplace=True)

#Divide total funding per pupil into 10 deciles
#print('Count of NC Districts by Totals Funds Per Student at Elementary School Level ')
#pd.cut(distData['lea_total_perpupil'], 10).value_counts().sort_index()

## Top Performing Districts with Lowest Average Per Pupil Funding - Elementary School Level 

In [32]:
#Find top performing districts in the lowest funding decile.
#Funds_distData = distData[distData['lea_total_perpupil'] < 9197.25]
#Funds_distData.sort_values('spg_score', ascending=False)